strawberry/types/fields/resolver.py

```diff
@@ -3,8 +3,9 @@
 import inspect
 import sys
 import warnings
+from asyncio import iscoroutinefunction
```

Comment:
This will not work for python 3.12

Check [here](https://github.com/django/asgiref/blob/main/asgiref/sync.py#L56) how asgiref does to support this for both 3.12 and older versions.

You can probably copy the:

```python
if hasattr(inspect, "markcoroutinefunction"):
    iscoroutinefunction = inspect.iscoroutinefunction
else:
    iscoroutinefunction = asyncio.iscoroutinefunction
```

In here and use it